community/kernel/src/main/java/org/neo4j/kernel/impl/factory/GraphDatabaseFacade.java

```diff
@@ -354,6 +354,12 @@ public Result execute( String query, Map<String,Object> parameters ) throws Quer
     {
         // ensure we have a tx and create a context (the tx is gonna get closed by the Cypher result)
         InternalTransaction transaction = beginTransaction( KernelTransaction.Type.implicit, AccessMode.Static.FULL );
+        return execute( transaction, query, parameters );
+    }
+
+    public Result execute( InternalTransaction transaction, String query, Map<String,Object> parameters )
```

Comment:
I think it is a good idea to add a comment here to explain that this `execute` is for testing only.